staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -662,6 +662,29 @@ func (qs *queueSet) cancelWait(req *request) {
 	qs.obsPair.RequestsWaiting.Add(-1)
 }
 
+func (qs *queueSet) canAccomodateSeatsLocked(seats int) bool {
+	switch {
+	case seats > qs.dCfg.ConcurrencyLimit:
+		// we have picked the queue with the minimum virtual finish time,
+		// but the number of seats this request asks for exceeds the concurrency limit
+		// TODO: this is a quick fix for now, once we have borrowing in place we will not need it
```

Comment:
What is described in the KEP is this concept of additional latency. Copying from the KEP:
```
One more important aspect to resolve is what happens if a given priority level doesn't have enough concurrency units assigned to it. To be on the safe side we should probably implement borrowing across priority levels. However, given we don't want to block introducing the width concept on design and implementation of borrowing, until this is done we have two main options:

cap the width at the concurrency units assigned to the priority level
reject requests for which we won't be able to allocate enough concurrency units
To avoid breaking some users, we will proceed with the first option (when computing the cap we should also report requests that we believe are too wide for a given priority level - it would allow operators to adjust configs). That said, to accommodate for the inaccuracy here we will introduce a concept of additional latency for a request. This basically means that after the request finishes in a real world, we still don't mark it as finished in the virtual world for additional latency. Adjusting virtual time of a queue to do that is trivial. The other thing to tweak is to ensure that the concurrency units will not get available for other requests for that time (because currently all actions are triggerred by starting or finishing some request). We will maintain that possibility by wrapping the handler into another one that will be sleeping for additional latence after the request is processed.

Note that given the estimation for duration of processing the requests is automatically corrected (both up and down), there is no need to change that in the initial version.
```

I don't necessary need this to be implemented in this PR, but let's just add a TODO to implement this (this isn't that hard to add and borrowing will take us a lot of time).

Reply:
BTW - I prototyped that (and couple other things) in: https://github.com/kubernetes/kubernetes/pull/102875
Feel free to pick up that PR too (though we should probably add additionallatency in a separate PR not in this one).

Reply:
thanks, I am happy to pick it up, yeah, i will do it in a separate PR.